src/features/payment/models/npgModel.ts

```diff
@@ -0,0 +1,29 @@
+import { FieldId } from "../components/IframeCardForm/types";
+
+enum NpgEvtDataErroCode {
+  HF0001 = "HF0001",
+  HF0002 = "HF0002",
+  HF0003 = "HF0003",
+  HF0004 = "HF0004",
+  HF0005 = "HF0005",
+  HF0006 = "HF0006",
+  HF0007 = "HF0007",
+  HF0009 = "HF0009",
+}
+export interface NpgEvtData {
+  id: FieldId;
+  errorCode: NpgEvtDataErroCode;
+  errorMessage: string;
+}
+
+export interface NpgFlowStateEvtData {
+  data: {
+    url: string;
+  };
+}
+
+export enum NpgFlowState {
+  REDIRECTED_TO_EXTERNAL_DOMAIN = "REDIRECTED_TO_EXTERNAL_DOMAIN",
+  PAYMENT_COMPLETE = "PAYMENT_COMPLETE",
+  READY_FOR_PAYMENT = "READY_FOR_PAYMENT",
+}
```

Comment:
Great!
But, I read in the Nexi API that the list of states are:
```
PAYMENT_METHOD_SELECTION
CARD_DATA_COLLECTION
READY_FOR_PAYMENT
REDIRECTED_TO_EXTERNAL_DOMAIN
GDI_VERIFICATION
PAYMENT_COMPLETE
```
Maybe you could map also the miss ones.

Reply:
Good idea! State added